src/coreclr/scripts/superpmi.py

```diff
@@ -831,6 +837,10 @@ async def run_pmi(print_prefix, assembly, self):
                 pmi_command_env = env_copy.copy()
                 pmi_complus_env = complus_env.copy()
                 pmi_complus_env["JitName"] = self.collection_shim_name
+
+                if self.coreclr_args.pmi_path is not None:
+                    root_env["PMIPATH"] = ";".join(self.coreclr_args.pmi_path)
```

Comment:
The problem with this is the `root_env` is modified, and will persist and be set in the subsequent use in crossgen2.

Maybe it's just simplest to set it at the top of this function, and let it be used/visible everywhere.

It would be more complex and maybe unnecessarily so to do something like:
```
new_root_env = os.environ.copy()
new_root_env["PMIPATH"] = ...
... use new_root_env instead of root_env ...
```

Reply:
I see, I didn't realize that we set do both `--pmi` and `--crossgen2` during the same run of `superpmi.py collect`